changes to comp_z

dopplerchase opened the pull request:
Code for issue #1559, where composite reflectivity code carried artifacts through with poor quality radar data from velocity sweeps. I also made a change to the interpolator because it was flagged as depreciated (e.g., scipy.interpolate.interp2d). 

- [X] Closes #1559 

This is now open for comment from the maintainers! Happy to make changes as needed. 

Example usage: 

``compz = pyart.retrieve.composite_reflectivity(radar, field="reflectivity")`` 

![image](https://github.com/ARM-DOE/pyart/assets/22037322/76b3e001-738f-4fbb-a960-199238ea4f04)


The code is flexible, and allows you to still have the old functionality through the simple bool: 

``compz = pyart.retrieve.composite_reflectivity(radar, field="reflectivity",same_nyquist=False)``

![image](https://github.com/ARM-DOE/pyart/assets/22037322/4ede664d-2307-4623-9043-1c58f29c56c8)

dopplerchase committed:
changes to comp_z

mgrover1 commented:
@dopplerchase - is there anything we can do to help here?

dopplerchase commented:
@mgrover1 find me some more time? lol We need to think carefully about Ryan's comments noted on https://github.com/ARM-DOE/pyart/issues/1559

mgrover1 commented:
no worries 😄 that is a good point!!

decadeneo commented:
I'm glad to hear that someone has encountered a similar bug. When directly using the function `pyart.retrieve.composite_reflectivity`, I found that the interpolated composite reflectivity values are all NaN. Upon diving into the source code, I was prompted to change the interpolation method. However, when attempting to modify it with alternative interpolation methods, I encountered an error indicating that the number of azimuth angles per layer in my radar data is inconsistent, thereby preventing a smooth interpolation process.

zssherman commented:
@mgrover1 I'm seeing the testing fail due to the depreciation of interp2d, we might need to do a PR just with the depreciation change. I can work on that.

zssherman committed:
Merge branch 'main' into composite_z_fix

zssherman committed:
Resolve merge conflict in comp_z.py

Removed conflicting code related to z interpolator configuration and application.

RBhupi commented:
It is failing multiple test on my local machine.
Merging `main` into `composite_z_fix` commit to auto-triggers CI.

RBhupi committed:
Merge branch 'main' into composite_z_fix

RBhupi committed:
FIX:linting errors

RBhupi committed:
FIX: comp_z same_nyquist review; add test)

RBhupi committed:
FIX:default same_nyquist=False so comp_z works without nyquist metadata
